core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2179,6 +2179,16 @@ private[spark] object Utils extends Logging {
       .getOrElse(UserGroupInformation.getCurrentUser().getShortUserName())
   }
 
+
+  // Returns the groups to which the current user belongs.
+  def getCurrentUserGroups(sparkConf: SparkConf, username: String): Set[String] = {
+    val groupMappingServiceProvider = Utils.classForName(sparkConf.get("spark.user.groups.mapping",
+      "org.apache.spark.security.ShellBasedGroupsMappingProvider")).newInstance.
+      asInstanceOf[org.apache.spark.security.GroupMappingServiceProvider]
```

Comment:
we should have a try catch around here incase users specifies a class that isn't found and the newInstance fails.  Just log an error if it fails.